internal/services/cdn/cdn_frontdoor_rule_set_association_resource.go

```diff
@@ -0,0 +1,245 @@
+	if err := setRouteRuleSetAssociations(d, meta, id, nil, props, "deleting", "waiting for the deletion of"); err != nil {
+		return err
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func validateRuleSetsAssociation(d *pluginsdk.ResourceData, meta interface{}, id *parse.FrontDoorRuleSetAssociationId, isCreate bool) error {
+	newRuleSets, newRuleSetsList, err := expandRuleSets(d.Get("cdn_frontdoor_rule_set_ids").(*pluginsdk.Set).List())
+	if err != nil {
+		return err
+	}
+
+	// Make sure the route exists and get the routes rule set association list...
+	routeRuleSetAssociations, _, err := getRouteRuleSetProperties(d, meta, id)
+	if err != nil {
+		return err
+	}
+
+	// Only validate the rule sets if there are rule sets defined in the association resource
+	if newRuleSets != nil {
+		// Make sure the rule sets exist...
+		if err = ruleSetExists(d, meta, newRuleSets); err != nil {
+			return err
+		}
+
+		// validate the new rule sets are associated with the route...
+		if len(*newRuleSets) != 0 {
+			notAssociated := make([]string, 0)
+
+			for _, v := range *newRuleSets {
+				if len(routeRuleSetAssociations) == 0 || !sliceContainsString(routeRuleSetAssociations, v.ID()) {
+					notAssociated = append(notAssociated, v.RuleSetName)
+				}
+			}
+
+			if len(notAssociated) != 0 {
+				return fmt.Errorf("the CDN FrontDoor Route(Name: %q) is currently not associated with the CDN FrontDoor Rule Sets: %s. Please remove the CDN FrontDoor Rule Sets from your configuration", id.AssociationName, strings.Join(notAssociated, ", "))
+			}
+		}
+
+		// on the initial creation of the resource we need to make sure ALL of the associated
+		// rule sets in the route resource are also included in the list of rule sets in the association...
+		if isCreate {
+			// validate that all of the routes associated rule sets are included in the association resource...
+			if len(routeRuleSetAssociations) != 0 {
+				notAssociated := make([]string, 0)
+
+				for _, v := range routeRuleSetAssociations {
+					if len(newRuleSetsList) == 0 || !sliceContainsString(newRuleSetsList, v.(string)) {
+						routeRuleSet, err := parse.FrontDoorRuleSetID(v.(string))
+						if err != nil {
+							return err
+						}
+
+						notAssociated = append(notAssociated, routeRuleSet.RuleSetName)
+					}
+				}
+
+				if len(notAssociated) != 0 {
+					return fmt.Errorf("the CDN FrontDoor Route(Name: %q) is currently associated with the CDN FrontDoor Rule Sets: %s. Please add the CDN FrontDoor Rule Sets to your configuration", id.AssociationName, strings.Join(notAssociated, ", "))
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func updateRuleSetsAssociations(d *pluginsdk.ResourceData, meta interface{}, id *parse.FrontDoorRuleSetAssociationId, errorTxt string, futureErrorTxt string) error {
+	newRuleSets := d.Get("cdn_frontdoor_rule_set_ids").(*pluginsdk.Set).List()
+
+	// first validate the resource
+	if err := validateRuleSetsAssociation(d, meta, id, false); err != nil {
+		return err
+	}
```

Comment:
as this seems like a no-op we should remove it and let the api handle it?

Reply:
Fixed.